packages/x-date-pickers/src/internals/models/props/toolbar.ts

```diff
@@ -18,9 +18,9 @@ export interface BaseToolbarProps<TValue> extends ExportedBaseToolbarProps {
   disabled?: boolean;
   readOnly?: boolean;
   // TODO v6: Remove
```

Comment:
I think that we agreed to stop following the MD guidelines no matter what and to focus an building the best pickers possible (picking MD ideas if they are good and dropping the one we don't like, like the Clock on desktop for instance).

For the Field mode switching, I had the feeling that we agreed to drop it.
But I may had misunderstood, and of course that is something we can re-implement.
My idea was to add an example on the doc on how to add it in user land.

Reply:
I'm personally not a fan of this input switching behaviour, although, it's a worthwhile compromise—at least until our picker interface allows for easy/quick year switching.
Because without input and our current mobile behaviour—it would be really painful to enter birthdate. 🤷 

Taking into consideration the above issue—I wouldn't mind keeping this mobile input toggling behaviour in v6 and consider dropping it once the UX of calendar behaviour is more refined. 🤔

Reply:
My end goal was more to tell to people to use the field for this kind of use case.
Because if the dev know that his use case is more keyboard-friendly than view-friendly, then I find it weird to force him to hide the keyboard behind an icon in the view.

I would be interested in the opinion of the rest of the team on this.

And I do agree that we can add the current behavior back if it's a question to complicate to answer during this alpha.

Reply:
For birthdays, the field, or either 3 inputs/select are better than any view.

When I enter my birthday or any date I know, the component can only catch typos.

Our components are helpful for picking a date you do not already know about. Because in such a case, it provides extra information:
- The date I'm going to pick is the first monday of the month
- It's the first available date

Reply:
@alexfauquette I do agree, but for the new pickers, do you think it is worth re-implementing the current input inside the view on mobile ?

Reply:
I added this topic to todays meeting

Reply:
No, I do not see the interest of having an input inside a view